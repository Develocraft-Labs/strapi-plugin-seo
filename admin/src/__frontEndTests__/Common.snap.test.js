/**
 * @jest-environment jsdom
 */
import React from "react";
import {
  BoxColumn,
  Column,
  EvenColumn,
  FlexWrapRow,
  Row,
} from "../components/ui/common";
import renderer from "react-test-renderer";

describe("Row component", () => {
  it("Should Render Row", () => {
    const tree = renderer.create(<Row>content</Row>).toJSON();

    expect(tree).toMatchSnapshot();
    expect(tree).toHaveStyleRule("display", "flex");
    expect(tree).toHaveStyleRule("flex-direction", "row");
  });
  it("Should Render FlexWrapRow", () => {
    const tree = renderer
      .create(
        <FlexWrapRow>
          <div>child</div>
        </FlexWrapRow>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
    expect(tree).toHaveStyleRule("display", "flex");
    expect(tree).toHaveStyleRule("flex-direction", "row");
    expect(tree).toHaveStyleRule("flex-wrap", "wrap");
    expect(tree).toHaveStyleRule("margin-left", "2rem", {
      modifier: "> *",
    });
  });
  it("Should Render Column", () => {
    const tree = renderer.create(<Column>content</Column>).toJSON();

    expect(tree).toMatchSnapshot();
    expect(tree).toHaveStyleRule("display", "flex");
    expect(tree).toHaveStyleRule("flex-direction", "column");
  });

  it("Should Render EvenColumn", () => {
    const tree = renderer.create(<EvenColumn>content</EvenColumn>).toJSON();

    expect(tree).toMatchSnapshot();
    // Check if it extends Column properly
    expect(tree).toHaveStyleRule("display", "flex");
    expect(tree).toHaveStyleRule("flex-direction", "column");
    expect(tree).toHaveStyleRule("justify-content", "space-evenly");
  });

  it("Should Render BoxColumn", () => {
    const tree = renderer.create(<BoxColumn>content</BoxColumn>).toJSON();

    expect(tree).toMatchSnapshot();
    expect(tree).toHaveStyleRule("background-color", "#ffffff");
    expect(tree).toHaveStyleRule("border-radius", "2px");
    expect(tree).toHaveStyleRule("box-shadow", "0 2px 4px #e3e9f3");
  });
});
