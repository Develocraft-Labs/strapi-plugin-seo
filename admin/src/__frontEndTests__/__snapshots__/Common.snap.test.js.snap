// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Row component Should Render BoxColumn 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c1 {
  padding: 2rem;
  background-color: #ffffff;
  border-radius: 2px;
  box-shadow: 0 2px 4px #e3e9f3;
}

<div
  className="c0 c1"
>
  content
</div>
`;

exports[`Row component Should Render Column 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

<div
  className="c0"
>
  content
</div>
`;

exports[`Row component Should Render EvenColumn 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c1 {
  -webkit-box-pack: space-evenly;
  -webkit-justify-content: space-evenly;
  -ms-flex-pack: space-evenly;
  justify-content: space-evenly;
}

<div
  className="c0 c1"
>
  content
</div>
`;

exports[`Row component Should Render FlexWrapRow 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c1 {
  padding-top: 2rem;
  margin-left: -2rem;
  margin-top: -2rem;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  background-color: transparent;
}

.c1 > * {
  margin-left: 2rem;
  margin-top: 2rem;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  min-width: 300px;
}

@media (max-width:768px) {
  .c1 {
    margin-left: 0;
  }

  .c1 > * {
    margin-left: 0;
    min-width: 100%;
    max-width: 100%;
  }
}

<div
  className="c0 c1"
>
  <div>
    child
  </div>
</div>
`;

exports[`Row component Should Render Row 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

<div
  className="c0"
>
  content
</div>
`;
